pkg/registry/apis/provisioning/repository_controller.go

```diff
@@ -310,12 +325,28 @@ func (rc *RepositoryController) process(item *queueItem) error {
 		}
 		if !res.Success {
 			logger.Error("repository is unhealthy", "errors", res.Errors)
+		} else {
+			logger.Info("repository is healthy")
+		}
+
+		patch, err := json.Marshal(map[string]any{
```

Comment:
@ryantxu  This is how I have fixed the issue with the health not being updated or overwritten.